### ISSUE TITLE
Futurepack Ships can't fly(teleport) pilot gets kicked, ship stays.

### ISSUE THREAD

**Issue by ProsperCraft** [SpongePowered/SpongeForge#2024]
**I am currently running**

- SpongeForge version: spongeforge-1.12.2-2611-7.1.0-BETA-2944
- Forge version: forge-1.12.2-14.23.2.2619-universal
- Java version: openjdk version "1.8.0_161"
- Operating System: Centos 7

- Plugins/Mods:
Plugins (14): Minecraft, Minecraft Coder Pack, SpongeAPI, SpongeForge, EconomyLite, LuckPerms, MMCReboot, MagiBridge, Nations, Nucleus, PayDay, PwnFilter Sponge Plugin, SleepVote, rankup

 Mods (168): Minecraft, Minecraft Coder Pack, Forge Mod Loader, Minecraft Forge, SpongeAPI, SpongeForge, /dank/null, AI Improvements, Applied Fluidics, Applied Llamagistics, Applied Llamagistics Core, Architect, Astikoor, Backpacks!, Bad Wither No Cookie! Reloaded, Bailey's Dailies, Bed Bugs, Better Animals Mod, Better Builder's Wands, Better Questing, Better Questing Quest Book, BiblioCraft, Binnie Core, Binnie's Botany, Binnie's Design, Binnie's Extra Bees, Binnie's Extra Trees, Binnie's Genetics, BiomeTweaker, Biomes O' Plenty, Bookshelf, Car Mod, Cavern II, Ceramics, Chameleon, ChinjufuMod, Chisel, Chisels & Bits, Chococraft 3, Chunk Pregenerator, CoFH Core, CoFH World, CodeChicken Lib, Cooking for Blockheads, Corail Pillar, Corail Pillar - Biomes O'Plenty Extension, Corail Pillar - Chisel Extension, Corail Pillar - Forestry Extension, CosmeticArmorReworked, CraftTweaker JEI Support, CraftTweaker2, CustomNPCs, Dark Roleplay Core, Dark Roleplay Medieval, Decocraft, DeepResonance, Default World Generator, Display, Dooglamoo Cities, Edible Bugs Mod, ElecCore, Elevator Mod, EnderTanks, EqualDragons, Extra Bit Manipulation, Farming for Blockheads, Farseek, Fast Leaf Decay, FoamFix, FoamFixCore, Forestry, Futurepack, Futurepack API, Futurepack Forestry Addon, Hydrophobia, Immersive Engineering, Immersive Petroleum, Immersive Railroading, Immersive Tech, Industrial Foregoing, Industrial Wires, IndustrialCraft 2, Inventory Tweaks, JEI Bees, Just Enough Items, LLibrary, LagGoggles, Landlust, Lapis Stays in the Enchanting Table, Laser Level, LibrarianLib, LibrarianLib Stage 2, Long Fall Boots, LunatriusCore, MTLib, MagicBees, Mantle, Mekanism, MekanismTools, Millenaire, MineColonies, MineTweakerRecipeMaker, Minecraft Transport Simulator, Mod Tweaker, MrCrayfish's Furniture Mod, MrCrayfish's Jumping Castle Mod, Multi-Beds, Multi-Storage, Natura, Nature's Compass, No FOV, NoNausea, NotEnoughIDs, OC Glasses, Open Glider, OpenComputers, OpenComputers (Core), OpenComputers Sensors, OpenEye, Pam's HarvestCraft, Personal Cars, Placebo, Platforms, Prefab, Pumpkin Carving, RC Roads, RF Lux, RandomPatches, Reborn Core, Redstone Flux, Rope Bridge, ServerObserver, Shadowfacts' Forgelin, ShetiPhian-Core, Signal-Catcher, Signpost, Standard Expansion, Steve's Carts 2, Storage Drawers, Streams, Super Sound Muffler, Tech Reborn, Techguns, Techguns Core, Tell Me, Tesla Core Lib, Tesla Core Lib Registries, Thermal Dynamics, Thermal Expansion, Thermal Foundation, TickProfiler, Tomb Many Graves, ToroChess, Track API, Tumbleweed, Unloader, Waddles, Waila, Wearables, What Are We Looking At, Wizardry, Xtones, Zetta Industries, careerbees, mxTune, p455w0rd's Library, ptrmodellib, �9RuneCraft

**Issue Description**
We are having a ton of changing world and teleport problems after adding Sponge, so I suspect this is related

Basically The user makes a futurepack ship, and keys in coords, then hits go, and its supposed to move the ship there, but it kicks the player instead.

https://gist.github.com/ProsperCraft/bb6c015cd22abab4733d567f004ee95f

[2018-02-25-7.log.gz](https://github.com/SpongePowered/SpongeForge/files/1757348/2018-02-25-7.log.gz)

**Comment by Zidane** [comment]
@ProsperCraft 

https://gist.github.com/ProsperCraft/bb6c015cd22abab4733d567f004ee95f#file-gistfile1-txt-L199

[FPPacketHandler$MessageRe                             */
questTerrain.class:?]

Not a Sponge issue, they are doing stuff off the main thread they shouldn't be. Just because it works in Forge doesn't mean its not wrong.

@gabizou Fix the printout of these logs, this is awful...

**Comment by mcenderdragon** [comment]
Why is calling "getBlockState" wrong outside the main thread ?

**Comment by me4502** [comment]
Block data isn’t stored in a thread safe manner. Grabbing that data on another thread means you could be grabbing it whilst it’s being modified.

**Comment by mcenderdragon** [comment]
Ok, has sponge fixed the Problem with the sheduled tasks for servers? The last time it crashed because it executed to many tasks at once.
